presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -878,7 +900,12 @@ public void dropTable(MetastoreContext metastoreContext, String databaseName, St
                     .stopOnIllegalExceptions()
                     .run("dropTable", stats.getDropTable().wrap(() ->
                             getMetastoreClientThenCall(metastoreContext, client -> {
+                                Table table = client.getTable(databaseName, tableName);
                                 client.dropTable(databaseName, tableName, deleteData);
+                                String tableLocation = table.getSd().getLocation();
+                                if (deleteFilesOnDrop && deleteData && isManagedTable(table) && !isNullOrEmpty(tableLocation)) {
+                                    deleteDirRecursive(hdfsContext, hdfsEnvironment, new Path(tableLocation));
+                                }
```

Comment:
Can we have the following code chunk to avoid overhead?
```java
if (!deleteFilesOnDrop) {
    client.dropTable(databaseName, tableName, deleteData);
}
else {
    Table table = client.getTable(databaseName, tableName);
    client.dropTable(databaseName, tableName, deleteData);
    String tableLocation = table.getSd().getLocation();
    if (deleteData && isManagedTable(table) && !isNullOrEmpty(tableLocation)) {
        deleteDirRecursive(hdfsContext, hdfsEnvironment, new Path(tableLocation));
    }
}

Reply:
Done